src/data_loader.py

```diff
@@ -0,0 +1,47 @@
+# src/data_loader.py
+
+import pandas as pd
+import logging
+from typing import Optional
+
+logger = logging.getLogger(__name__)
+
+
+def load_data(
+    good_file_path: str,
+    promo_file_path: str,
+    nrows: Optional[int] = None,
+    shuffle: bool = True,
+) -> pd.DataFrame:
+    """
+    Load promotional and non-promotional text data from CSV files and combine them into a single dataset.
+
+    Args:
+        good_file_path (str): Path to the CSV file containing non-promotional text data.
+        promo_file_path (str): Path to the CSV file containing promotional text data.
+        nrows (int, optional): Number of rows to read from each CSV file. Defaults to None, meaning all rows are read.
+        shuffle (bool, optional): Whether to shuffle the combined dataset. Defaults to True.
+
+    Returns:
+        pd.DataFrame: A pandas DataFrame containing text data and corresponding labels (1 for promotional, 0 for non-promotional).
+    """
+    logger.info("Loading promotional and non-promotional data.")
+    try:
+        good_df = pd.read_csv(good_file_path, nrows=nrows)
+        promo_df = pd.read_csv(promo_file_path, nrows=nrows)
+    except FileNotFoundError as e:
+        logger.error(f"Error reading files: {e}")
+        raise
+
+    logger.info("Assigning labels to data.")
+    good_df["label"] = 0
```

Comment:
If we need the label at other places in the code too, we could create an enum for that. But we can skip it for now and see whether we need it later.

Reply:
Might be interesting in the future. I would skip it for now until we know how it could help then